dotnet/samples/Concepts/Filtering/AzureOpenAI_DeploymentSwitch.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Azure.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+namespace Filtering;
+
+/// <summary>
+/// This sample shows how to switch between Azure OpenAI deployments based on the functions that are being called.
+/// This can be useful if semantic caching is enabled and you want to switch to a different deployment based on the functions that are being called.
+/// </summary>
+public class AzureOpenAI_DeploymentSwitch(ITestOutputHelper output) : BaseTest(output)
+{
+    [Fact]
+    public async Task DeploymentSwitchAsync()
+    {
+        Assert.NotNull(TestConfiguration.AzureOpenAI.ChatDeploymentName);
+        Assert.NotNull(TestConfiguration.AzureOpenAI.Endpoint);
+
+        // Create a logging handler to output HTTP requests and responses
+        using var httpHandler = new HttpClientHandler();
+        using var loggingHandler = new LoggingHandler(httpHandler, this.Output);
+        using var httpClient = new HttpClient(loggingHandler);
+
+        // Create KernelBuilder with an auto function invocation filter
+        var kernelBuilder = Kernel.CreateBuilder();
+        kernelBuilder.Services.AddSingleton<IAutoFunctionInvocationFilter>(new AutoFunctionInvocationFilter(this.Output));
+
+        // Define the endpoints for the two Azure OpenAI services
+        var endpoint1 = "https://lightspeed-team-shared-openai-eastus.openai.azure.com/";
+        var endpoint2 = "https://lightspeed-team-shared-openai-swedencentral.openai.azure.com/";
+
```

Comment:
Suggest moving to variables all the other changing details, like deploymentName, regio, similar to how was done with `endpoints`.